demisto_sdk/tests/integration_tests/format_integration_test.py

```diff
@@ -2110,3 +2133,528 @@ def test_verify_deletion_from_conf_script_format_with_deprecate_flag(
+                    "type": "shortText",
+                    "name": "Alias Incident Field"
+                }
+            ],
+            "marketplaces": [
+                "marketplacev2"
+            ]
+        }
+    )
+    alias_incident_field_content = {
+            "cliName": "aliasincidentfield",
+            "id": "incident_aliasincidentfield",
+            "name": "Alias Incident Field",
+            "marketplaces": [
+                "marketplacev2", "xsoar"
+            ]
+        }
+    alias_incident_field = pack.create_incident_field(
+        name="alias_incident_field",
+        content=alias_incident_field_content
+    )
+
+    result = runner.invoke(main, [FORMAT_CMD, "-i", incident_field.path, "-at", "-y", "-ngr"])
+    message = f"Skipping formatting of marketplaces field of aliases for {incident_field.path}" \
+              f" as the no-graph argument was given."
+    assert result.exit_code == 0
+    assert not result.exception
+    assert str_in_call_args_list(logger_info.call_args_list, message)
+
+    # get_dict_from_file returns a tuple of 2 object. The first is the content of the file,
+    # the second is the type of the file.
+    file_content = get_dict_from_file(alias_incident_field.path)[0]
+    assert file_content.get('marketplaces', []) == ["marketplacev2", "xsoar"]
+
+
+def test_format_removing_fields_not_in_content_from_mapper(mocker, monkeypatch, repo):
+    """
+    Given
+    - A mapper.
+
+    When
+    - Running format command on it
+
+    Then
+    -  Ensure that the unknown field was removed from the mapper.
+    """
+    logger_info = mocker.patch.object(logging.getLogger("demisto-sdk"), "info")
+    monkeypatch.setenv("COLUMNS", "1000")
+    incident_field_name = "Unknown Incident Field"
+    mapper_content = {
+            "description": "",
+            "feed": False,
+            "id": "Mapper - Incoming Mapper",
+            "mapping": {
+                "Mapper Finding": {
+                    "dontMapEventToLabels": True,
+                    "internalMapping": {
+                        incident_field_name: {
+                            "simple": "Item"
+                        }
+                    }
+                },
+            },
+            "name": "Mapper - Incoming Mapper",
+            "type": "mapping-incoming",
+            "version": -1,
+            "fromVersion": "6.5.0"
+        }
+
+    runner = CliRunner()
+    pack = repo.create_pack("PackName")
+    mapper = pack.create_mapper(
+        name="mapper",
+        content=mapper_content
+    )
+
+    # Mock the behavior of Neo4jContentGraphInterface
+    class MockedContentGraphInterface:
+        def __enter__(self):
+            return self
+
+        def __exit__(self):
+            pass
+
+        def search(self, object_id):
+            # Simulate the graph search
+            if object_id == "Mapper - Incoming Mapper":
+                return [MockedMapperNode()]
+            return []
```

Comment:
can we use a fixture instead of redefining the `MockedSomething` class over and over again?

Reply:
It is very complicated. In each test I get a different type of content item that needs to have the information about the item, including it's path, content etc. For that I need to create the Items with the repo.create_pack(), and than create the needed content item. The usage of repo requires that everything will be in a function, which is a problem because it's multiple classes.